components/raftstore/src/store/peer.rs

```diff
@@ -1439,16 +1491,13 @@ where
             }
             CheckApplyingSnapStatus::Success => {
                 self.post_pending_read_index_on_replica(ctx);
+                // If there is a snapshot, it must belongs to the last ready.
+                self.raft_group
+                    .on_persist_ready(self.last_unpersisted_number);
             }
             CheckApplyingSnapStatus::Idle => {}
         }
 
-        if !self.pending_messages.is_empty() {
-            fail_point!("raft_before_follower_send");
```

Comment:
I'm afraid they may be used in external tests.

Reply:
I change them back.